### ISSUE TITLE
Restrict stage visibility based on membership

### ISSUE THREAD

**Issue by 3mcd** [pubpub/platform#762]
## Motivation

So users only see stages they have permission to view.

## Requirements

Modify stage-related queries so that:
- Community contributors can only see stages they have been added to directly or via groups.
- Community editors and admins can always see all stages.

If a user does not have the ability to see any stages, the link to "Workflows" in the side nav should not appear.

## Acceptance Criteria

- Uses functions implemented in #671 to determine if the user has the `viewStage` capability for each rendered stage.
- End-to-end tests that assert that stages on the workflows page are only visible when the above requirements are met.

**Comment by 3mcd** [comment]
@kalilsn I agree that the requirement to show/hide the Workflows link adds a bit of overhead (one query per page load) for little gain. Instead I think we could always show the link and just put an informative message on the Workflows page if no stages are accessible to the user. Thoughts @gabestein ?

**Comment by allisonking** [comment]
this would involve the following, though I'd love confirmation that this sounds on the right track:

* modifying this query to filter by only stages the user has the `viewStage` capability for https://github.com/pubpub/platform/blob/76f44f265abffce1430511a35cec493dec17cc79/core/lib/server/stages.ts#L52-L103
* if the query returns an empty list in the Workflows page here, then render a message that there are no accessible stages https://github.com/pubpub/platform/blob/76f44f265abffce1430511a35cec493dec17cc79/core/app/c/%5BcommunitySlug%5D/stages/manage/page.tsx#L69
* will returning only some stages mess with the graph layout? maybe it'll just work?

**Comment by gabestein** [comment]
> will returning only some stages mess with the graph layout? maybe it'll just work

I think this is an editor- and admin-only view anyway, so we can probably just not render it if you don't have access to view all stages?
